skill-tree /src/main/java/com/RDS/skilltree/EndorsementList/EndorsementListService.java

```diff
@@ -0,0 +1,52 @@
+package com.RDS.skilltree.EndorsementList;
+
+import com.RDS.skilltree.Endorsement.EndorsementModel;
+import com.RDS.skilltree.Endorsement.EndorsementRepository;
+import com.RDS.skilltree.Exceptions.NoEntityException;
+import com.RDS.skilltree.User.UserModel;
+import com.RDS.skilltree.User.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class EndorsementListService {
+    @Autowired
+    private final EndorsementListRepository endorsementListRepository;
+    private final EndorsementRepository endorsementRepository;
+    private final UserRepository userRepository;
+
+    public EndorsementListService(EndorsementListRepository endorsementListRepository, EndorsementRepository endorsementRepository, UserRepository userRepository) {
+        this.endorsementListRepository = endorsementListRepository;
+        this.endorsementRepository = endorsementRepository;
+        this.userRepository = userRepository;
+    }
+
+
+    public EndorsementListModel createEndorsementListEntry(EndorsementListDRO endorsementListDRO) {
+        EndorsementListModel endorsementListEntry = new EndorsementListModel();
+
+        UUID endorserId = endorsementListDRO.getEndorserId();
+        UUID endorsementId = endorsementListDRO.getEndorsementId();
+        Optional<UserModel> endorserOptional = userRepository.findById(endorserId);
+        Optional<EndorsementModel> endorsementOptional = endorsementRepository.findById(endorsementId);
+        if (endorserOptional.isPresent() && endorsementOptional.isPresent()) {
+
+            endorsementListEntry.setEndorser(endorserOptional.get());
+            endorsementListEntry.setEndorsement(endorsementOptional.get());
+            endorsementListEntry.setDescription(endorsementListDRO.getDescription());
+            endorsementListEntry.setType(endorsementListDRO.getType());
+            endorsementListRepository.save(endorsementListEntry);
+            return endorsementListEntry;
+
+        } else {
+            if (endorserOptional.isEmpty())
+                throw new NoEntityException("User with id:" + endorserId + "not found");
```

Comment:
This seems to be an error as there are two `throw new` statements. Did you mean to throw `NoEntityException` for no endorsement?
Also, in the error message try to have spaces after the strings for concatenation. Otherwise for id: 123 your message will look like `endorsement with id:123not found`

Reply:
no the endorser is checked as empty here hence user not found exception